### PR TITLE
Add io.github.bkueng.qMasterPassword

### PR DESCRIPTION
**Pull request by bkueng**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.bkueng.qMasterPassword.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    post-install:
+      - mkdir -p /app/share/icons/hicolor/512x512/apps
+      - cp /app/share/pixmaps/qmasterpassword.png /app/share/icons/hicolor/512x512/apps/$FLATPAK_ID.png
+      - cp /app/share/metainfo/{qMasterPassword,$FLATPAK_ID}.appdata.xml
+      - sed -i "s/<id>qMasterPassword.desktop<\/id>/<id>$FLATPAK_ID<\/id>/g" /app/share/metainfo/$FLATPAK_ID.appdata.xml
+      - cp /app/share/applications/{qMasterPassword,$FLATPAK_ID}.desktop
+      - sed -i "s/Icon=qmasterpassword/Icon=$FLATPAK_ID/g" /app/share/applications/$FLATPAK_ID.desktop
```
Comment (bbhtt) [inline io.github.bkueng.qMasterPassword.yml:32]:
```suggestion
      - cp /app/share/metainfo/{qMasterPassword,$FLATPAK_ID}.appdata.xml
      - sed -i "s/<id>qMasterPassword.desktop<\/id>/<id>$FLATPAK_ID<\/id>/g" /app/share/metainfo/$FLATPAK_ID.appdata.xml
      - cp /app/share/applications/{qMasterPassword,$FLATPAK_ID}.desktop
      - sed -i "s/Icon=qmasterpassword/Icon=$FLATPAK_ID/g" /app/share/applications/$FLATPAK_ID.desktop
```

Use rename-desktop-file, rename-appdata-file and rename-icon, see man flatpak-manifest. They will fix the id in appdata, icon in desktop etc. automatically.

Comment (bbhtt) [inline io.github.bkueng.qMasterPassword.yml:32]: Use rename-desktop-file, rename-appdata-file and rename-icon, see man flatpak-manifest. They will fix the id in appdata, icon in desktop etc. automatically.

Comment (hfiguiere) [reply io.github.bkueng.qMasterPassword.yml:32]: and then you don't need to use sed on the .destkop

File: io.github.bkueng.qMasterPassword.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      - type: archive
+        url: https://github.com/technion/libscrypt/archive/v1.22.tar.gz
+        sha256: a2d30ea16e6d288772791de68be56153965fe4fd4bcd787777618b8048708936
+  - name: qMasterPassword
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    post-install:
+      - mkdir -p /app/share/icons/hicolor/512x512/apps
+      - cp /app/share/pixmaps/qmasterpassword.png /app/share/icons/hicolor/512x512/apps/$FLATPAK_ID.png
```
Comment (hfiguiere) [inline io.github.bkueng.qMasterPassword.yml:28]:
```suggestion
      - install -Dm644 /app/share/pixmaps/qmasterpassword.png -t /app/share/icons/hicolor/512x512/apps
```

(and then `rename-icon`)

File: io.github.bkueng.qMasterPassword.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      - cp /app/share/metainfo/{qMasterPassword,$FLATPAK_ID}.appdata.xml
+      - sed -i "s/<id>qMasterPassword.desktop<\/id>/<id>$FLATPAK_ID<\/id>/g" /app/share/metainfo/$FLATPAK_ID.appdata.xml
```
Comment (hfiguiere) [inline io.github.bkueng.qMasterPassword.yml:30]: don't use sed

```suggestion
      - appstream-util modify $FLATPAK_DEST/share/metainfo/qMasterPassword.appdata.xml id ${FLATPAK_ID},
```

Comment (hfiguiere) [reply io.github.bkueng.qMasterPassword.yml:30]: (sadly this is not handled by flatpak-builder)

**Comment by hfiguiere** [comment]
bot, build io.github.bkueng.qMasterPassword

**Comment by bkueng** [comment]
Thanks for the hints and the quick review. I updated accordingly, and it looks like I don't have to use `appstream-util modify $FLATPAK_DEST/share/metainfo/qMasterPassword.appdata.xml id ${FLATPAK_ID}`

File: io.github.bkueng.qMasterPassword.yml (hunk 3 above)
Comment (bbhtt) [reply io.github.bkueng.qMasterPassword.yml]: Builder handles this if you use rename desktop file

**Comment by bbhtt** [comment]
bot, build io.github.bkueng.qMasterPassword

**Comment by bkueng** [comment]
Ah yes there's an optional functionality requiring X11. I disabled that and it works for me on Wayland as well now.

**Comment by bbhtt** [comment]
The app is using a different bus name than the FLATPAK_ID https://github.com/search?q=repo%3Abkueng%2FqMasterPassword%20org.bkueng.qMasterPassword&type=code

**Comment by hfiguiere** [comment]
> The app is using a different bus name than the FLATPAK_ID https://github.com/search?q=repo%3Abkueng%2FqMasterPassword%20org.bkueng.qMasterPassword&type=code
> 
> It seems to launch but this might be problematic https://docs.flatpak.org/en/latest/flatpak-command-reference.html please see the session bus policy section. Ideally the bus name should match the flatpak id and the app id too.

it should be fixed upstream. Using domains you don't control for dbus is as bad as for app-id, but no one checks.

**Comment by bkueng** [comment]
I changed the DBus ID to match the flatpak id `io.github.bkueng.qMasterPassword`

**Comment by hfiguiere** [comment]
bot, build io.github.bkueng.qMasterPassword

**Comment by bkueng** [comment]
Thanks!

**Pull request closed without merging**